stacks-signer/src/client/stacks_client.rs

```diff
@@ -358,6 +364,103 @@ impl StacksClient {
         self.get_account_entry(address).map(|entry| entry.nonce)
     }
 
+    /// Get information about the tenures between `chosen_parent` and `last_sortition`
+    pub fn get_tenure_forking_info(
+        &self,
+        chosen_parent: &ConsensusHash,
+        last_sortition: &ConsensusHash,
+    ) -> Result<Vec<TenureForkingInfo>, ClientError> {
+        let mut tenures = self.get_tenure_forking_info_step(chosen_parent, last_sortition)?;
+        if tenures.is_empty() {
+            return Ok(tenures);
+        }
+        while tenures.last().map(|x| &x.consensus_hash) != Some(chosen_parent) {
+            let new_start = tenures.last().ok_or_else(|| {
+                ClientError::InvalidResponse(
+                    "Should have tenure data in forking info response".into(),
+                )
+            })?;
+            let mut next_results =
+                self.get_tenure_forking_info_step(chosen_parent, &new_start.consensus_hash)?;
+            if next_results.is_empty() {
+                return Err(ClientError::InvalidResponse(
+                    "Could not fetch forking info all the way back to the requested chosen_parent"
+                        .into(),
+                ));
+            }
+            // SAFETY check: next_results isn't empty, because of the above check. otherwise, remove(0) could panic.
+            next_results.remove(0);
```

Comment:
You could avoid the potential panic by reversing `next_results`, `pop`-ing an item, and then reversing it again.

Reply:
I just replaced with VecDeques which lets you fallibly `pop_front()`.